### DISCUSSION TITLE
Step by step instructions for Windows 10/11 (ARM and Intel)

### DISCUSSION THREAD

**Discussion by litewrap** [AparokshaUI/Adwaita#12] [Improvements]
I would like to try Adwaita for Swift on Windows 10/11 both ARM and Intel.
I am familiar with Swift on Windows 10/11 but not at installing the Adwaita stuff for Windows.
Can you provide step by step install instructions ?

**Comment by david-swift** [comment]
I'm very sorry for not having answered your question for so long, and unfortunately, I'm still not able to answer it.

I didn't try this package on Windows, but knew from other projects, [Rnote](https://github.com/flxzt/rnote) and [Gaphor](https://github.com/gaphor/gaphor), that Libadwaita would work on Windows. When I tried to use it on Windows, it turned out to be difficult to get it building.

Rnote's build instructions for Windows are [well-documented](https://github.com/flxzt/rnote/blob/main/misc/building/rnote-windows-build.md). They use [MSYS2](https://www.msys2.org/) (which is the way also recommended by the GTK docs), but I couldn't get Swift running inside this system and I was [not the first person to do so](https://forums.swift.org/t/swift-via-msys2-mingw-w64/47155/2).

Some other approaches are listed in the [gvsbuild repo](https://github.com/wingtk/gvsbuild?tab=readme-ov-file#gvsbuild). I might try some of them once I'm less busy, or you can (note to install Libadwaita 1.5, not only GTK).

The easiest way to get it work is to install a Linux system such as [Fedora Workstation](https://fedoraproject.org/workstation/) and dual-boot with Windows, it doesn't need too much disk space, or use a VM. The advantage here is [Flatpak](https://flatpak.org/) that allows building apps without installing a single dependency on the system (except for Flatpak). Also, Libadwaita integrates best with the GNOME desktop environment. 

I'll try the other approaches in the gvsbuild repo in the next weeks and then, I'll hopefully be able to provide a satisfactory answer and detailed instructions.
